cmd/fyne/internal/commands/build.go

```diff
@@ -280,18 +280,21 @@ func createMetadataInitFile(srcdir string, app *appData) (func(), error) {
 	}
 	defer metadataInitFile.Close()
 
-	err = templates.FyneMetadataInit.Execute(metadataInitFile, app)
-	if err == nil {
-		iconResName := "fyneMetadataIcon"
-		if app.icon != "" {
-			writeResource(app.icon, iconResName, metadataInitFile)
-		} else {
-			v := fmt.Sprintf("var %s = (fyne.Resource)(nil)\n", iconResName)
-			_, err = metadataInitFile.Write([]byte(v))
-			if err != nil {
-				fyne.LogError("Error writing icon placeholder", err)
-			}
+	app.ResGoString = "nil"
+	if app.icon != "" {
```

Comment:
It feels like this is the wrong way round - should an icon specified on the command line not override the one in the metadata?

Reply:
I am not sure I follow. If the `app.icon` fields points to a file, we load the file and embed the icon data. This seems to me like it is the same behaviour as before or am I missing something?